### PR TITLE
Update organizeDeclarations rule to correctly parse more cases of instancePropertyWithBody

### PR DESCRIPTION
**Pull request by calda**
This PR updates the new `organizeDeclarations` rule (#701) to correctly parse more instance of `instancePropertyWithBody`. The previous heuristic for this was incorrect, and was incorrectly classifying all of these examples as `instanceProperty` (without body):

```swift
var withBody3: String = "bar" {
    didSet { print("didSet") }
}

var withBody4: String = "bar" {
    didSet { print("didSet") }
}

var withBody5: () -> String = { "bar" } {
    didSet { print("didSet") }
}

var withBody6: String = { "bar" }() {
    didSet { print("didSet") }
}
```

### CODE REVIEW

File: Tests/ParsingHelpersTests.swift [hunk 1]
```diff
@@ -426,6 +426,46 @@ class ParsingHelpersTests: XCTestCase {
...
+    }
+
+    func testParameterBodyAfterExecutedClosureIsNotClosure() {
+        let formatter = Formatter(tokenize("""
+        var withBody6: String = { "bar" }() {
+            didSet { print("didSet") }
+        }
+        """))
+
+        XCTAssertFalse(formatter.isStartOfClosure(at: 19))
```
Comment (calda) [inline Tests/ParsingHelpersTests.swift:466]: `isStartOfClosure(at:)` is incorrectly reporting `true` here (for the brace before the `didSet`). 

I'm having a hard time finding a fix (it calls into `lastSignificantKeyword(at:)`, which is returning `nil` here).

File: Tests/RulesTests.swift [hunk 2]
```diff
@@ -14311,4 +14319,55 @@ class RulesTests: XCTestCase {
...
+            }
+
+            var withBody6: String = { "bar" }() {
+                didSet { print("didSet") }
+            }
+
+        }
+        """
+
+        testFormatting(
```
Comment (calda) [inline Tests/RulesTests.swift:14368]: This test currently fails because of https://github.com/nicklockwood/SwiftFormat/pull/716/files#r478797017

File: Sources/Rules.swift [hunk 3]
```diff
@@ -5156,10 +5156,14 @@ public struct _FormatRules {
...
-                    [
-                        $0.markComment(from: "%c"),
+                    Array(Set([
+                        // The user's specific category separator template
                         $0.markComment(from: formatter.options.categoryMarkComment),
-                    ]
+                        // Other common variants that we would want to replace with the correct variant
+                        $0.markComment(from: "%c"),
+                        $0.markComment(from: "// MARK: %c"),
+                        $0.markComment(from: "// MARK: - %c"),
```
Comment (calda) [inline Sources/Rules.swift:5165]: Bonus unrelated change, I added `// MARK: - %c` as another common variant to replace with the correct category separator template .

I found a lot of examples in our codebase where folks were using this variant. Adding it to our list of common variants makes sure we replace it with the correct `formatter.options.categoryMarkComment`, rather than adding a duplicate comment).

File: Tests/ParsingHelpersTests.swift (hunk 1 above)
Comment (nicklockwood) [reply Tests/ParsingHelpersTests.swift:466]: Sorry, I only noticed this when it was too late in the release process to fix it. I'll find a solution and fix it in the next release. It's obscure enough that it hopefully won't be a big problem in the meantime,

Comment (calda) [reply Tests/ParsingHelpersTests.swift:466]: Thanks :)
